### PR TITLE
Wait for host before killing process

### PR DESCRIPTION
**Pull request by millerds**
Thank you for your pull request!  Please provide the following information.

---

**Change Description**:
When Excel tests complete we need to close the workbook before the host, but we kill the process to quickly.  We need to wait for Excel (and other hosts) before killing the process.

1. **Do these changes impact any *npm scripts* commands (in package.json)?** (e.g., running 'npm run start')
    No.

2. **Do these changes impact *VS Code debugging* options (launch.json)?**
    No.

3. **Do these changes impact *template output*?** (e.g., add/remove file, update file location, update file contents)
    No.

4. **Do these changes impact *documentation*?** (e.g., a tutorial on https://docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins)
    No.

If you answered yes to any of these please do the following:
    > Include 'Rick-Kirkham' in the review
    > Make sure the README file is correct

**Validation/testing performed**:

    Ran automated tests multiple times.

### CODE REVIEW

**Review by igor-ribeiiro — APPROVED** [review]

File: test/end-to-end/src/test-helpers.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ export async function closeDesktopApplication(application: string): Promise<bool
       throw new Error(`${application} is not a valid Office desktop application.`);
   }
 
+  await sleep(3000); // wait for host to settle
```
Comment (akrantz) [inline test/end-to-end/src/test-helpers.ts:30]: Why is this necessary?

Comment (millerds) [reply test/end-to-end/src/test-helpers.ts:30]: In Excel specifically we were at one point calling close workbook, but not closing Excel and this would leave the app open after tests.  It was changed to close Excel after calling close workbook, but it seems that the closing of Excel is happening too quickly after calling close workbook and the next time Excel is opened the recovery pane (or on mac a dialog) is opened.  By putting the pause here we are allowing any final host operations (like close workbook) to happen before killing the process and avoiding any recovery states.
